script.js

```diff
@@ -0,0 +1,841 @@
+            timer = setTimeout(tick, 50);
+          }, 500);
+        });
+        keyDom.addEventListener("mouseup", () => {
+          clearTimeout(timer);
+          isMouseUp = true;
+          if (isAnimationEnd) keyDom.classList.remove("keyboard-line__button_pressed");
+        });
+        keyDom.addEventListener("mouseleave", () => {
+          clearTimeout(timer);
+          isMouseUp = true;
+          if (isAnimationEnd) keyDom.classList.remove("keyboard-line__button_pressed");
+        });
+        keyDom.addEventListener("animationend", () => {
+          isAnimationEnd = true;
+          if (isMouseUp) keyDom.classList.remove("keyboard-line__button_pressed");
+        });
+        vKeyBoard.focusTextArea();
+      }
+    }
+
+    keyboardLine.appendChild(keyDom);
+  });
+});
+
+const layoutMessage = document.createElement("div");
+layoutMessage.innerText = "To change layout use Shift + Alt (developed on Windows OS)";
+layoutMessage.className = "keyboard__message";
+keyboard.appendChild(layoutMessage);
+
+document.addEventListener("keydown", (event) => {
+  const key = document.querySelector(`div[key=${event.code}]`);
+  if (key) {
+    if (event.code === "CapsLock") {
+      const capsDom = document.querySelector("div[key=\"CapsLock\"");
+      if (capsDom.classList.contains("keyboard-line__button_pressed")) {
+        capsDom.classList.remove("keyboard-line__button_pressed");
+        FUNCTIONAL_KEYS.caps = false;
+      } else {
+        capsDom.classList.add("keyboard-line__button_pressed");
+        FUNCTIONAL_KEYS.caps = true;
+      }
+    } else key.classList.add("keyboard-line__button_pressed");
+    if (event.code === "Tab") {
+      event.preventDefault();
+      vKeyBoard.insertSymbol("\t");
+    } else if (event.code === "AltLeft" || event.code === "AltRight") event.preventDefault();
+    if (
+      (FUNCTIONAL_KEYS.caps && event.key.toLowerCase() === event.key)
+      || (!FUNCTIONAL_KEYS.caps && event.key.toUpperCase() === event.key)
+    ) {
+      event.preventDefault();
+      vKeyBoard.insertSymbol(
+        FUNCTIONAL_KEYS.caps ? event.key.toUpperCase() : event.key.toLowerCase(),
+      );
+    }
+  }
+});
+
+document.addEventListener("keyup", (event) => {
+  const key = document.querySelector(`div[key=${event.code}]`);
+  if (key) {
+    if (event.code !== "CapsLock") key.classList.remove("keyboard-line__button_pressed");
+  }
+});
+
+function changeLayout() {
+  const rows = document.querySelectorAll(".keyboard-panel > .keyboard-line");
+  [...rows].forEach((row, i) => {
+    const buttons = row.querySelectorAll(".keyboard-line__button");
+    buttons.forEach((button, j) => {
+      const key = arrayOfButtonKeys[i][j];
+      if (getType(key) === "number" || getType(key) === "letter") {
+        button.classList.forEach((c) => {
+          if (c !== "keyboard-line__button") button.classList.remove(c);
+        });
+        const mainText = button.querySelector(
+          ".keyboard-line__button-main-text",

```

Comment:
For this strings it's better to create some object of constants, because you can make some mistake in ther word, and it will be really hard to find the mistake.

Reply:
All these strings was moved to styleConstants.js

https://github.com/BlackTruth/virtual-keyboard/blob/6297a0817c03affb1948c9e73a6352687e6ff449/src/styleConstants.js#L1-L11